tools/evaluation/calculate_coco_metrics.py

```diff
@@ -91,6 +91,25 @@ def calculate_metrics(pred_dir, gt_dir, bbox_format):
     cocoEval.summarize()
 
 
+    # Calculate mAP for each class:
+    for key, value in class_name_to_id.items():
```

Comment:
This is a code duplication from the above. This should be a function call:

def calculate_map(gt_anns, pred_anns, class_name_to_id=None)

And you trigger it twice: above and within this loop